### PR TITLE
utreexo: new Rust crate

### PR DESCRIPTION
**Pull request by bobg**
Based on the Go implementation at github.com/bobg/utreexo.

### CODE REVIEW

File: utreexo/Cargo.lock [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# This file is automatically @generated by Cargo.
```
Comment (oleganza) [inline utreexo/Cargo.lock]: Libraries generally do not put Cargo.lock under version control. This is only useful for applications to lock down all dependencies. Libraries can be used with differently locked down applications, so they only specify versions per SemVer rules to allow upgrades.

File: utreexo/src/update.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+use super::hash::Hash;
+use super::proof::{Proof, ProofStep};
+use super::utreexo::Utreexo;
+
+use std::collections::HashMap;
+
+pub struct Update {
+    pub(crate) u: Utreexo,
```
Comment (oleganza) [inline utreexo/src/update.rs]: nitpick: could we have more spelled out names, because `u.u.` and `hh` in the code are kinda hard to read :-)

**Comment by eupn** [comment]
Didn't know that there is a crate here, so I've ended up doing the same work 🤦‍♂: https://github.com/eupn/utreexo-rs

I will maintain this crate on crates.io so it may be unnecessary to embed utreexo crate into a project but use it as a Cargo.toml dependency.

**Comment by oleganza** [comment]
Closed since we are using implementation in #308

**Pull request closed without merging**